bayes_opt/target_space.py

```diff
@@ -37,6 +41,8 @@ def __init__(self, target_func, pbounds, constraint=None, random_state=None):
             optionally specify a seed for a random number generator
         """
         self.random_state = ensure_rng(random_state)
+        self._allow_duplicate_points = allow_duplicate_points
```

Comment:
I think this should be explicitly mutable, so I would not prepend the underscore.

Reply:
you think we should enable this to be changed post instantiation? 
since the behavior is inside `Optimizer._space` which is itself private, then would have to add some sort of setting method to the main `Optimizer` method?

Reply:
Fair enough. Let's leave it like this then.